### PR TITLE
Handle backendURLs without protocol

### LINKED ISSUE #102: Managed API Invoke fails if backendURL doesn't have a protocol defined
Currently the gateway doesn't handle `backendUrl`'s without the protocol prefix (`http://` or `https://`).

### PR DESCRIPTION
**Pull request by alexsong93**
Fixes #102

### CODE REVIEW

File: api-gateway-config/scripts/lua/routing.lua [hunk 1]
```diff
@@ -59,10 +59,15 @@ function _M.processCall()
       end
       -- Parse backend url
       local u = url.parse(opFields.backendUrl)
+      -- add http:// if no protocol is specified
+      if u.host == nil then
```
Comment (codymwalker) [inline api-gateway-config/scripts/lua/routing.lua]: It looks like host is the domain, where scheme is the protocol. So wouldn't we need to look for scheme here?

https://github.com/golgote/neturl#url-parser

Comment (alexsong93) [inline api-gateway-config/scripts/lua/routing.lua]: @codymwalker ah good catch - it's been fixed. Also fixed a little bug in the management interface for updating APIs with `/` basePath.

File: api-gateway-config/scripts/lua/routing.lua [hunk 2]
```diff
@@ -59,10 +59,15 @@ function _M.processCall()
       end
       -- Parse backend url
       local u = url.parse(opFields.backendUrl)
+      -- add http:// if no protocol is specified
+      if u.scheme == nil then
+        u = url.parse(utils.concatStrings({'http://', opFields.backendUrl}))
+      end
       ngx.req.set_uri(getUriPath(u.path))
       ngx.var.backendUrl = opFields.backendUrl
```
Comment (codymwalker) [inline api-gateway-config/scripts/lua/routing.lua]: @alexsong93 It looks that we're still storing the backendUrl here without the added scheme. I'm not sure that's a problem though as the only other place I saw `ngx.var.backendUrl` used was in mapping. Is there anywhere else we're using this variable where we would want/need the scheme attached?

Comment (alexsong93) [inline api-gateway-config/scripts/lua/routing.lua]: @codymwalker yup I saw that as well. You're right the only other place `ngx.var.backendUrl` is being used is in mapping, and it doesn't need the scheme. It's being used there to parse out the query (not sure if it's being done the best way, but that's probably for another PR..). I tested out the query mappings and it seems to work fine.

**Pull request merged**